pyfstat/mcmc_based_searches.py

```diff
@@ -283,28 +283,17 @@ def logp(self, theta_vals, theta_prior, theta_keys, search):
         return np.sum(H)
 
     def logl(self, theta, search):
+        in_theta = copy.copy(self.fixed_theta)
         for j, theta_i in enumerate(self.theta_idxs):
-            self.fixed_theta[theta_i] = theta[j]
```

Comment:
It seemed really ugly to overwrite `self.fixed_theta` in each iteration, so I cleaned this up along the way, here and for several child classes.

Reply:
Is fixed_theta a dictionary? Why not just `self.fixed_theta.copy()`?

Reply:
It should be a dict, but as elsewhere: future work.